### ISSUE TITLE
reservations example fails

### ISSUE THREAD

**Issue by elinaawise** [apache/accumulo-examples#45]
>reserve room06 20140101 alice bob eve mallory trent

```java
2019-04-18 10:23:09,006 [examples.reservations.ARS] WARN : Could not write to the ConsoleReader.
java.lang.IllegalStateException: This client was closed.
	at org.apache.accumulo.core.clientImpl.ClientContext.ensureOpen(ClientContext.java:114)
	at org.apache.accumulo.core.clientImpl.ClientContext.createConditionalWriter(ClientContext.java:539)
	at org.apache.accumulo.examples.reservations.ARS.reserve(ARS.java:97)
	at org.apache.accumulo.examples.reservations.ARS$1.run(ARS.java:276)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by ctubbsii** [comment]
@elinaawise Just a quick note: you can use triple-backtick to format a block on GitHub issues. See https://github.github.com/gfm/#fenced-code-blocks for reference.

**Comment by Manno15** [comment]
I took a look at this and it appears the reason why the client is closing is because the try-block around the client initialization closes right after connecting. 

What would be the best solution? I have tried some reformatting of ARS.Java but with the way it is structured currently to take in user arguments makes that a bit more complicated of a solution. I could also separate the connect feature to a function on its own so the other functions (like reserve) can check on connection status and then reconnect if need be. That seems more like a workaround than a clean solution so I am hesitant on doing that. Since this is an example, we could remove the try-block for the client connection and then call client.close() somewhere else if required. Any other ideas?

**Comment by Manno15** [comment]
@mikewalch Any thoughts on this?

**Comment by keith-turner** [comment]
> I took a look at this and it appears the reason why the client is closing is because the try-block around the client initialization closes right after connecting.

I took a look, that does seem to be the problem.  Below is the code you mentioned.

https://github.com/apache/accumulo-examples/blob/master/src/main/java/org/apache/accumulo/examples/reservations/ARS.java#L304

>  Since this is an example, we could remove the try-block for the client connection

That would be my choice.  Its an example, so do not want to do anything that makes it hard to follow. Could add a comment saying  `its good to close clients when possible, but it was not done here for simplicity`

**Comment by Manno15** [comment]
Thanks @keith-turner for taking a look at it. I will make a pull request with the changes
